src/appshell/view/preferences/accessibilitypreferencesmodel.cpp

```diff
@@ -0,0 +1,87 @@
+/*
+ * SPDX-License-Identifier: GPL-3.0-only
+ * MuseScore-CLA-applies
+ *
+ * MuseScore
+ * Music Composition & Notation
+ *
+ * Copyright (C) 2021 MuseScore BVBA and others
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+#include "accessibilitypreferencesmodel.h"
+
+#include "ui/internal/themeconverter.h"
+
+using namespace mu::appshell;
+using namespace mu::ui;
+
+AccessibilityPreferencesModel::AccessibilityPreferencesModel(QObject* parent)
+    : QObject(parent)
+{
+}
+
+QVariantList AccessibilityPreferencesModel::highContrastThemes() const
+{
+    QVariantList result;
+
+    for (const ThemeInfo& theme : uiConfiguration()->themes()) {
+        if (theme.codeKey == HIGH_CONTRAST_BLACK_THEME_CODE || theme.codeKey == HIGH_CONTRAST_WHITE_THEME_CODE) {
+            result << ThemeConverter::toMap(theme);
+        }
+    }
+
+    return result;
+}
+
+int AccessibilityPreferencesModel::currentThemeIndex() const
+{
+    ThemeList hcThemes;                                        //creation of hcThemes ensures that we only search in
+                                                               //a list of High Contrast Themes and not the entire list of Themes
+    for (const ThemeInfo& theme : uiConfiguration()->themes()) {
+        if (theme.codeKey == HIGH_CONTRAST_BLACK_THEME_CODE || theme.codeKey == HIGH_CONTRAST_WHITE_THEME_CODE) {
+            hcThemes.push_back(theme);
+        }
+    }
+
+    for (int i = 0; i < static_cast<int>(hcThemes.size()); ++i) {
+        if (hcThemes[i].codeKey == (uiConfiguration()->currentTheme()).codeKey) {
+            return i;
+        }
+    }
+
+    return -1;
+}
+
+void AccessibilityPreferencesModel::setCurrentThemeIndex(int index)
+{
+    ThemeList hcThemes;
+
+    for (const ThemeInfo& theme : uiConfiguration()->themes()) {
+        if (theme.codeKey == HIGH_CONTRAST_BLACK_THEME_CODE || theme.codeKey == HIGH_CONTRAST_WHITE_THEME_CODE) {
+            hcThemes.push_back(theme);
+        }
+    }
```

Comment:
I don't think adding those inline functions in `uitypes.h` would be of much help...
There aren't many moments in the logic where I need just the ThemeCodes of the themes, and I'm afraid that introducing more and more variables with the word "theme" will decrease code comprehension (it was already hard keeping track of all the "Theme" related variables and what they do 😅 )

> ```using ThemeCodeList = std::vector<ThemeCode>;```

From what I understand, adding this bit of code will replace my `if (theme.codeKey == HIGH_CONTRAST_BLACK_THEME_CODE || theme.codeKey == HIGH_CONTRAST_WHITE_THEME_CODE)` statement, and allow me to directly use a for loop to iterate over the desired themes based on their CodeKey inside this vector.

However, I still want to raise the same point that creating more variables will be confusing later on. I think it is more expressive and direct the way it currently is. I have used that loop twice in the code, but if you feel that it is still redundant, or if I find that I need to use it more than twice, I can shift that logic to a private function in the same file and call that wherever I need for a bit more tidiness :)

(Also, something very similar to what you suggested already exists in uitypes.h i.e. `using ThemeList = std::vector<ThemeInfo>;`
This is logically very different of course, but very similar in naming)

Reply:
That said though, I see what you mean when you say that this would optimize certain areas of the code. 🤔

Reply:
If we use consistent naming, then I think it should not be too complicated. a `Theme` is a struct with all the information for a theme, a `ThemeCode` is only a key for a theme. That's doable, isn't it? 
But I agree that things become unclear when the naming is not consistent. For example, `uiConfiguration()->setCurrentTheme(...)` takes a `ThemeCode`, so it is named wrongly. Feel free to adjust that. 

With a function for getting `highContrastThemeCodes`, you could simplify for example `AccessibilityPreferencesModel::setCurrentThemeIndex` like this:
```cpp
void AccessibilityPreferencesModel::setCurrentThemeIndex(int index)
{
    ThemeCodeList hcThemeCodes = highContrastThemeCodes();
    if (index < 0 || index >= static_cast<int>(hcThemeCodes.size())) {
        return;
    }

    if (hcThemeCodes[index] == uiConfiguration()->currentThemeCodeKey) {
        return;
    }

    uiConfiguration()->setCurrentTheme(hcThemes[index].codeKey);
    emit themesChanged();
}
```
Also `AccessibilityPreferencesModel::currentThemeIndex()` and the corresponding methods of AppearancePreferencesModel can be simplified.

Reply:
I have implemented something akin to this in this form:
```c++
bool UiTheme::isCurrentThemeHighContrast() const
{
    return configuration()->currentTheme().codeKey == HIGH_CONTRAST_BLACK_THEME_CODE
           || configuration()->currentTheme().codeKey == HIGH_CONTRAST_WHITE_THEME_CODE;
}

bool UiTheme::isCurrentThemeGeneral() const
{
    return configuration()->currentTheme().codeKey == LIGHT_THEME_CODE || configuration()->currentTheme().codeKey == DARK_THEME_CODE;
}
```

This is different from what you were suggesting, since the use-cases are different, but this still saves me a lot of time whenever i need a particular feature active only for certain themes. :)